src/main/java/com/bandwidth/sqs/queue/RetryingSqsQueue.java

```diff
@@ -0,0 +1,85 @@
+package com.bandwidth.sqs.queue;
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.sqs.model.AmazonSQSException;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiPredicate;
+
+import io.reactivex.Completable;
+import io.reactivex.Single;
+import io.reactivex.subjects.CompletableSubject;
+import io.reactivex.subjects.SingleSubject;
+
+/**
+ * The SqsRequestSender is only able to retry non-batched actions, since it doesn't have access to the individual
+ * actions inside of a batch. This class retries each individual action
+ */
+public class RetryingSqsQueue<T> implements SqsQueue<T> {
+
+    private final SqsQueue<T> delegate;
+    private final int retryCount;
+
+    public RetryingSqsQueue(SqsQueue<T> delegate, int retryCount) {
+        this.delegate = delegate;
+        this.retryCount = retryCount;
+    }
+
+
+    @Override
+    public Completable deleteMessage(String receiptHandle) {
```

Comment:
comment explaining that the actions which are batched are the ones that are retried here